### PR TITLE
enhance: improve rootcoord task scheduling policy

### PR DESCRIPTION
**Pull request by SimFG**
- issue: #30301

### CODE REVIEW

File: internal/rootcoord/create_collection_task.go [hunk 1]
```diff
@@ -658,3 +658,10 @@ func (t *createCollectionTask) Execute(ctx context.Context) error {
 
 	return undoTask.Execute(ctx)
 }
+
+func (t *createCollectionTask) GetLockerKey() LockerKey {
+	return NewLockerKeyChain(
+		NewClusterLockerKey(false),
+		NewDatabaseLockerKey(t.Req.GetDbName(), true),
```
Comment (xiaofan-luan) [inline internal/rootcoord/create_collection_task.go:665]: can we do craete collection with 
NewDatabaseLockerKey(t.Req.GetDbName(), false),
NewCollectionLockerKey(t.Req.GetDbName(), true)?

Same with drop collection

Comment (xiaofan-luan) [inline internal/rootcoord/create_collection_task.go:665]: Why can't create collection only host collection level write lock?

**Review by xiaofan-luan — COMMENTED** [review]
Good job!
very excited about this!

**Comment by xiaofan-luan** [comment]
@SimFG just let this pr so @wangting0128 could test on the concurrent DDL case

**Comment by xiaofan-luan** [comment]
I don't have a strong reason we can clean the cache on collection created

**Comment by SimFG** [comment]
rerun go-sdk

**Comment by SimFG** [comment]
rerun ut

**Comment by SimFG** [comment]
rerun ut

**Pull request merged**